### PR TITLE
Add Ansible playbooks

### PR DESCRIPTION
**Pull request by sneridagh**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,4 +5,6 @@ This is the repository used in Plone 6 demo in 6.demo.plone.org.
 You can find the backend in the `/api` folder, and the frontend in the root of the repo.
 You can find also convenience scripts in the supplied `Makefile`.
 
+For convenience, the deployment Ansible playbooks are included in `ansible` folder. They install the entire required dependencies and infrastructure, and the services as well.
```
Comment (stevepiercy) [inline README.md:8]:
```suggestion
For convenience the deployment Ansible playbooks are included in the `ansible` folder.
 They install the entire required dependencies and infrastructure as well as the services.
```

File: README.md [hunk 2]
```diff
@@ -5,4 +5,6 @@ This is the repository used in Plone 6 demo in 6.demo.plone.org.
...
+
 For any questions, please contact the @plone/ai-team
```
Comment (stevepiercy) [inline README.md:10]: Although not part of this PR, I have no idea how to contact `@plone/ai-team`. Is that an email address, Twitter handle, other? Please be explicit.

File: ansible/Vagrantfile [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    # ansible.limit = "<name_of_your_host>"
+    # ansible.tags = "yourtag"
+  end
+
+  # config.vm.provider "virtualbox" do |v|
+  #   v.memory = 4000
+  #   v.cpus = 2
+  # end
+
+  # This is needed for change port configs
```
Comment (stevepiercy) [inline ansible/Vagrantfile:32]:
```suggestion
  # This is needed to change port configs
```

File: ansible/Vagrantfile [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+  # Create a forwarded port mapping which allows access to a specific port
+  # within the machine from a port on the host machine. In the example below,
+  # accessing "localhost:8080" will access port 80 on the guest machine.
+  # config.vm.network "forwarded_port", guest: 80, host: 8080
+
+  # Create a private network, which allows host-only access to the machine
+  # using a specific IP.
+  # config.vm.network "private_network", ip: "192.168.33.10"
+
+  # Create a public network, which generally matched to bridged network.
```
Comment (stevepiercy) [inline ansible/Vagrantfile:50]:
```suggestion
  # Create a public network, which generally matches to a bridged network.
```

File: ansible/ansible.cfg [hunk 5]
```diff
@@ -0,0 +1,5 @@
+[defaults]
+inventory = hosts
+# additional paths to search for roles in, colon separated
```
Comment (stevepiercy) [inline ansible/ansible.cfg:3]:
```suggestion
# additional paths in which to search for roles, colon-separated
```

File: ansible/etc/nginx/template-volto-api-dev.conf [hunk 6]
```diff
@@ -0,0 +1,63 @@
+# This defines in which IP and port Plone is running.
```
Comment (stevepiercy) [inline ansible/etc/nginx/template-volto-api-dev.conf:1]:
```suggestion
# This defines on which IP and port Plone is running.
```

File: ansible/etc/nginx/template-volto-api-dev.conf [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+server {
+    listen 443 ssl http2;
+    server_name {{ server_canonical_name }};
+    client_max_body_size 1G;
+
+    access_log /var/log/nginx/{{ server_canonical_name }}.access.log;
+    error_log /var/log/nginx/{{ server_canonical_name }}.error.log;
+
+    # ssl on;
+    # cloudflare takes care...
```
Comment (stevepiercy) [inline ansible/etc/nginx/template-volto-api-dev.conf:27]:
```suggestion
    # Cloudflare handles the secure certificate
```

File: ansible/tasks/template-volto-api.yml [hunk 8]
```diff
@@ -0,0 +1,81 @@
+---
+# Volto + Plone API template:
+# service_path: the server directory
+# api_port: The api port in the server
```
Comment (stevepiercy) [inline ansible/tasks/template-volto-api.yml:4]:
```suggestion
# api_port: The API port on the server
```

File: ansible/tasks/template-volto-api.yml [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+# volto_port: Volto port in the server
```
Comment (stevepiercy) [inline ansible/tasks/template-volto-api.yml:5]:
```suggestion
# volto_port: Volto port on the server
```

File: ansible/tasks/template-volto-api.yml [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+# service_name: The PM2 file name in the server (no extensions)
```
Comment (stevepiercy) [inline ansible/tasks/template-volto-api.yml:6]:
```suggestion
# service_name: The PM2 file name on the server (no extensions)
```

File: ansible/tasks/template-volto-api.yml [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+# service_api_instances: Array or string of instance names
+# htpassword_name: The password for protect the site. Leave in blank for no password
```
Comment (stevepiercy) [inline ansible/tasks/template-volto-api.yml:8]:
```suggestion
# htpassword_name: The password to protect the site. Leave blank for no password.
```

File: ansible/tasks/template-volto-api.yml [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+# server_canonical_name: The server DNS
```
Comment (stevepiercy) [inline ansible/tasks/template-volto-api.yml:9]:
```suggestion
# server_canonical_name: The server's name in the DNS entry.
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]
Thank you for this contribution!

In this pass, I reviewed comments and documentation only, and provided some feedback.

Is this deployment supposed to be on a local development environment or on a third party provider (such as Digital Ocean), both, or other? We should be explicit about any prerequisites and the environments where this playbook may be used.

Depending on the answer to that question, I will attempt a deployment to see how it actually works, and provide additional comments in a second review.

File: README.md (hunk 2 above)
Comment (sneridagh) [reply README.md:11]: It's the GitHub handler, @ericof there's a mail for the aiteam?

File: ansible/Vagrantfile (hunk 4 above)
Comment (sneridagh) [reply ansible/Vagrantfile:50]: Not mine, just test from the Vagrant init script.

**Comment by sneridagh** [comment]
> Is this deployment supposed to be on a local development environment or on a third party provider (such as Digital Ocean), both, or other? We should be explicit about any prerequisites and the environments where this playbook may be used.

This is for production deployment only. I hope this fills some gaps for people who never deployed Volto (or any Node app). For local development, there are other methods (documented in the docs) and now depending on what you want to accomplish. The installers team is also working into them but it's true that documentation should be improved.

**Comment by stevepiercy** [comment]
The documentation about the process to get this running does not exist. I tried to push a commit to this branch with my notes to add to the README.md, but I cannot push to this repo. I've pasted my notes below.

I assume that the process described in https://github.com/plone/ansible-playbook#plone-ansible-playbook applies for this playbook. If not please correct me.

----------------------------

## Ansible Usage

Checkout and work with this repository.

```shell
git clone https://github.com/plone/6.demo.plone.org.git
cd 6.demo.plone.org.git
# git checkout ansible # for testing this branch only
```

Create a Python virtual environment, upgrade its package management tools, and install Ansible into it.

```shell
python3 -m venv env
env/bin/pip install --upgrade pip setuptools
env/bin/pip install ansible
```

Create or reuse an existing SSH key to deploy into a virtual machine in the next step.

```shell
ssh-keygen
```

Create a remote virtual machine for production deployment, such as a Digital Ocean droplet or AWS EC2 instance.


[//]: # (I created a DO droplet of 1GB RAM, and I expect it to fail.)
[//]: # (The droplet might need to be resized.)

This playbook has been tested on Ubuntu 20.04 LTS only.
The minimum requirements for the virtual machine include ___GB memory.
Obtain the hostname of your virtual machine.
This process is left to the provider to document.

Update the Ansible configuration file at `ansible/vbox_host.cfg` with your hostname, replacing the value of `<name_of_your_host>`.

Run the playbook.

```shell
env/bin/ansible-playbook -i ansible/vbox_host.cfg ansible/volto-demo.yml

PLAY [all] *************************************************************************************************************

TASK [Gathering Facts] *************************************************************************************************
fatal: [143.198.238.54]: UNREACHABLE! => {"changed": false, "msg": "Failed to connect to the host via ssh: ssh: connect to host 127.0.0.1 port 2222: Connection refused", "unreachable": true}

PLAY RECAP *************************************************************************************************************
143.198.238.54             : ok=0    changed=0    unreachable=1    failed=0    skipped=0    rescued=0    ignored=0
```

*sad trombone*

**Comment by sneridagh** [comment]
@stevepiercy I just gave you access, sorry, I forgot to add proper permissions on creation, please feel free to push whatever you want to it. I would greatly appreciate it! Sorry if this contribution does not have a sense of completeness but I thought it's better to have it (although un-explained in a narrative way) than not having nothing.

I'm fighting in a lot of fronts nowadays, I'll try to get back to it. I'm preparing some things for the WPD, related to this.

**Comment by stevepiercy** [comment]
@sneridagh thank you for access!

I pushed my commit to this branch. Most of it is my best guess. I was not able to deploy successfully. If you can fill in any missing pieces, I would appreciate it.

https://github.com/plone/6.demo.plone.org/pull/1/commits/ec665a628827ee505333a113b4c1430bc9bcc7f2

**Comment by sneridagh** [comment]
Merging for convenience, since we have to rethink the whole deployment for the demo and use current deployment best practices.
